src/common/ownsql.h

```diff
@@ -72,8 +78,15 @@ class OCSYNC_EXPORT SqlQuery
 {
     Q_DISABLE_COPY(SqlQuery)
 public:
+    explicit SqlQuery() = default;
     explicit SqlQuery(SqlDatabase &db);
-    explicit SqlQuery(const QString &sql, SqlDatabase &db);
+    explicit SqlQuery(const QByteArray &sql, SqlDatabase &db);
+    /**
+     * Prepare the SQLQuery if it was not prepared yet.
+     * Otherwise, clear the results and the bindings.
+     * return false if there is an error
+     */
+    bool init(const QByteArray &sql, SqlDatabase &db);
```

Comment:
This function seems really convenient but the name is confusing. `prepareOnce`? `prepareOrClearResults`?

Now there are three ways of using this
```
SqlQuery q1;
q1.init();

SqlQuery q2(db);
q2.prepare();

SqlQuery q3(sql, db);
```
Can we get rid of one of them? Or document these three modes? Also move `prepare()` up here so the relation is clear visually.

Reply:
Maybe `initOrReset`  ?

I was thinking removing the constructor taking just a database, and making the `prepare() ` call private, but then there is even more code that would need to be adjusted.  Do you think I should?

Reply:
Yes, I think that'd make it easier to see how the query is supposed to be used.